src/ArcxAnalyticsSdk.ts

```diff
@@ -227,12 +248,18 @@ export class ArcxAnalyticsSdk {
   }
 
   private _trackSigning() {
-    const provider = getWeb3Provider()
-    if (!provider) {
+    if (!this.provider) {
       return false
     }
-    const request = provider.request
-    provider.request = async ({ method, params }: RequestArguments) => {
+
+    if (!this._originalRequest) {
+      this._originalRequest = this.provider.request
+    }
+
+    // Deliberately not using this._original request to not intefere with the transaction tracking's
+    // request modification
+    const request = this.provider.request.bind(this.provider)
```

Comment:
Same as above

Reply:
how do we lose link to request if provider is binded and has proper reference?